From d4eb3ab036f8c37c5bc5f45ad0fa4dc34b7228c8 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Wed, 24 Feb 2016 14:37:13 +0800
Subject: [PATCH 3/3] yaffs2: using new ->follow_link() and ->put_link()
 calling conventions

As what we did in commit 680baacbca69 ("new ->follow_link() and
->put_link() calling conventions").

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 fs/yaffs2/yaffs_vfs.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/fs/yaffs2/yaffs_vfs.c
+++ b/fs/yaffs2/yaffs_vfs.c
@@ -1060,7 +1060,7 @@ static int yaffs_readlink(struct dentry
 }
 
 #if (YAFFS_NEW_FOLLOW_LINK == 1)
-static void *yaffs_follow_link(struct dentry *dentry, struct nameidata *nd)
+static const char *yaffs_follow_link(struct dentry *dentry, void **cookie)
 {
 	void *ret;
 #else
@@ -1082,7 +1082,7 @@ static int yaffs_follow_link(struct dent
 		goto out;
 	}
 #if (YAFFS_NEW_FOLLOW_LINK == 1)
-	nd_set_link(nd, alias);
+	*cookie = alias;
 	ret = alias;
 out:
 	if (ret_int)
@@ -1114,9 +1114,9 @@ static void yaffs_put_inode(struct inode
 #endif
 
 #if (YAFFS_NEW_FOLLOW_LINK == 1)
-void yaffs_put_link(struct dentry *dentry, struct nameidata *nd, void *alias)
+void yaffs_put_link(struct inode *inode, void *cookie)
 {
-	kfree(alias);
+	kfree(cookie);
 }
 #endif
 
